app/controllers/staff/transactions_controller.rb

```diff
@@ -81,6 +82,17 @@ def id
     params[:id]
   end
 
+  def activity
+    @activity ||= Activity.find(activity_id)
+  end
+
+  def associate_transaction_to_submission
+    organisation = activity.organisation
+    fund = activity.fund? ? activity : activity.parent_activities.select { |a| a.level = :fund }
+    submission = Submission.active.where(organisation: organisation, fund: fund).first
```

Comment:
How about `find_by` instead of `where().first`?

Reply:
One day I will also learn how to code